### PR TITLE
Devcontainer: enable coredumps

### PR DESCRIPTION
**Pull request by thanodnl**
Add configuration for coredumps and document how to make sure they are enabled.

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,9 +1,12 @@
 {
-    "image": "ghcr.io/citusdata/citus-devcontainer:main",
+    "image": "ghcr.io/citusdata/citus-devcontainer:devcontainer-coredumps",
```
Comment (thanodnl) [inline .devcontainer/devcontainer.json:2]: TODO: before merging, revert image tag

**Review by eaydingol — APPROVED** [review]
Looks good to me. I successfully obtained the core-dump file and opened it using the added extension.

File: DEVCONTAINER.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+## Coredumps
+When postgres/citus crashes, there is the option to create a coredump. This is useful for debugging the issue. Coredumps are enabled in the devcontainer by default. However, not all environments are configured correctly out of the box. The most important configuration that is not standardized is the `core_pattern`. The configuration can be verified from the container, however, you cannot change this setting from inside the container as the filesystem containing this setting is in read only mode while inside the container.
+
+To verify if corefiles are written run the following command in a terminal. This shows the filename pattern with which the corefile will be written.
+```bash
+cat /proc/sys/kernel/core_pattern
+```
+
+This should be configured with a relative path or simply a simple filename. When your environment shows an absolute path you will need to change this setting. How to change this setting depends highly on the underlying system as the setting needs to be changed on the kernel of the host running the container.
```
Comment (JelteF) [inline DEVCONTAINER.md:11]:
```suggestion
This should be configured with a relative path or simply a simple filename, such as `core`. When your environment shows an absolute path you will need to change this setting. How to change this setting depends highly on the underlying system as the setting needs to be changed on the kernel of the host running the container.
```

File: DEVCONTAINER.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# Devcontainer
+
+## Coredumps
+When postgres/citus crashes, there is the option to create a coredump. This is useful for debugging the issue. Coredumps are enabled in the devcontainer by default. However, not all environments are configured correctly out of the box. The most important configuration that is not standardized is the `core_pattern`. The configuration can be verified from the container, however, you cannot change this setting from inside the container as the filesystem containing this setting is in read only mode while inside the container.
```
Comment (JelteF) [inline DEVCONTAINER.md:4]: `core_uses_pid` is another setting that matters, would be good to include the same wording for that.

**Pull request merged**